ccv/chains/evm/deployment/v1_7_0/changesets/create2_factory/deploy_create2_factory.go

```diff
@@ -27,6 +27,15 @@ func applyDeployCREATE2Factory(e cldf_deployment.Environment, cfg DeployCREATE2F
 		return cldf_deployment.ChangesetOutput{}, fmt.Errorf("chain with selector %d not found in environment", cfg.ChainSel)
 	}
 
+	// Require that the nonce of the deployer key is 0
+	nonce, err := evmChain.Client.PendingNonceAt(e.GetContext(), evmChain.DeployerKey.From)
+	if err != nil {
+		return cldf_deployment.ChangesetOutput{}, fmt.Errorf("failed to get nonce of deployer key: %w", err)
+	}
+	if nonce != 0 {
+		return cldf_deployment.ChangesetOutput{}, fmt.Errorf("nonce of the deployer key must be 0, got %d", nonce)
```

Comment:
nit: explaining why would be nice because not knowing how create2 works, this check feels wrong/strange.